File: chatait-backend-vue/src/router/modules/finance.ts
/*
 * Copyright 2023 Anlity <leo@leocode.net>. All rights reserved.
 * Use of this source code is governed by a AGPL v3.0 style
 * license that can be found in the LICENSE file.
 */

import { shallowRef } from 'vue'
import { MoneyCircleIcon } from 'tdesign-icons-vue-next'

export default [
  {
    path: '/finance',
    name: 'finance',
    component: () => import('@/layouts/index.vue'),
    redirect: '/finance/wallet-list',
    meta: { title: '财务管理', icon: shallowRef(MoneyCircleIcon) },
    children: [
      {
        path: 'wallet-list',
        name: 'financeWalletList',
        component: () => import('@/pages/finance/walletList.vue'),
        meta: { title: '钱包列表' },
      },
      {
        path: 'wallet-flow-list-balance',
        name: 'financeWalletFlowListBalance',
        component: () => import('@/pages/finance/walletFlowList.vue'),
        meta: { title: '余额流水' },
      },
      {
        path: 'wallet-flow-list-gpt3',
        name: 'financeWalletFlowListGpt3',
        component: () => import('@/pages/finance/walletFlowList.vue'),
        meta: { title: `Gpt3流水` },
      },
      {
        path: 'wallet-flow-list-gpt4',
        name: 'financeWalletFlowListGpt4',
        component: () => import('@/pages/finance/walletFlowList.vue'),
        meta: { title: `Gpt4流水` },
      },
      {
        path: 'wallet-flow-list-midjourney',
        name: 'financeWalletFlowListMidjourney',
        component: () => import('@/pages/finance/walletFlowList.vue'),
        meta: { title: `Midjourney流水` },
      },
    ],
  },
]

File: chatait-backend-server/app/model/response/finance.go
// Copyright 2023 Anlity <leo@leocode.net>. All rights reserved.
// Use of this source code is governed by a AGPL v3.0 style
// license that can be found in the LICENSE file.

package response

type FinanceWalletItem struct {
	UserId     string `json:"user_id"`
	Username   string `json:"username"`
	Nickname   string `json:"nickname"`
	Balance    int    `json:"balance"`
	Gpt3       int    `json:"gpt3"`
	Gpt4       int    `json:"gpt4"`
	Midjourney int    `json:"midjourney"`
}

type FinanceWalletList []*FinanceWalletItem

type FinanceWalletFlowItem struct {
	Id         string `json:"id"`
	UserId     string `json:"user_id"`
	Username   string `json:"username"`
	Nickname   string `json:"nickname"`
	Amount     int    `json:"amount"`
	Total      int    `json:"total"`
	IsIncr     int    `json:"is_incr"`
	TargetType string `json:"target_type"`
	TargetId   string `json:"target_id"`
	Remark     string `json:"remark"`
	AdminName  string `json:"admin_name"`
	CreatedAt  int    `json:"created_at"`
}

type FinanceWalletFlowList []*FinanceWalletFlowItem
